specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/applicationGateway.json

```diff
@@ -2076,6 +2430,148 @@
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private link resource."
+        }
+      },
+      "description": "Properties of Private Link Resource of an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkResource": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkResourcePropertiesFormat",
+          "description": "Properties of the application gateway private link resource."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the private link resource that is unique within an Application Gateway."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "PrivateLink Resource configuration of an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationGatewayPrivateLinkResource"
+          },
+          "description": "List of private link resources of an application gateway."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of results."
+        }
+      },
+      "description": "Response for ListApplicationGatewayPrivateLinkResources API service call. Gets all private link resources for an application gateway"
+    },
+    "ApplicationGatewayPrivateEndpointConnectionPropertiesFormat": {
+      "properties": {
+        "privateEndpoint": {
+          "readOnly": true,
+          "$ref": "./privateEndpoint.json#/definitions/PrivateEndpoint",
+          "description": "The resource of private end point."
+        },
+        "privateLinkServiceConnectionState": {
+          "$ref": "./privateLinkService.json#/definitions/PrivateLinkServiceConnectionState",
+          "description": "A collection of information about the state of the connection between service consumer and provider."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private endpoint connection resource."
+        },
+        "linkIdentifier": {
```

Comment:
Is it specific for application gateway?

Reply:
Didnt quiet follow the comment. 
Are you asking the property linkIdentifier in specific or privateEndpointConnection.

Reply:
I don't see same field for other service's private endpoint like storage, keyvault or acr.

Reply:
I have taken reference from privateLinkService. PrivateLinkService exposes the property on privateEndpointConnection. Please see privateLinkService.json 

Also its good for customers to know this link identifier in future when we support proxy protocol in order to corelate where the ip is coming from.

@malopMSFT did review the output for privateEndpoint connection and said its correct. 

He has suggested 1 more properties for privateLinkResources(just for consistenct nothing blocker), which I will add later as NRP object doesnot have it.